feyton/atlp-backend
# [#180969915] - Add tests and Fix module vulnerabilities

#### What does this PR do?
##### This PR fixes:
- Hanging async requests that raise error outside try/catch block
- Hanging request when a user try to logout after deleting account
- Returning wrong status for Email taken conflict @Jkadhuwa 

##### This PR adds:
- Continuous integrations tests using Mocha
- A more maintainable and modular code-base
- Error and requests logging using system file 
- Refined documentation with swagger

#### Description of Task to be completed?
This PR make the project ready for deployment to the main branch. With the test suite being added, deployment become easier. 

#### How should this be manually tested?
To test this PR, all needed is to:
1. **Clone this repository**
2. Follow the following commands to setup local environment

```
git clone https://github.com/feyton/atlp-backend.git 
git checkout refactoring
npm install
npm run dev
```
Make sure you have an .env file with a minimum of the following.

```
ACCESS_TOKEN_SECRET="Your access token secret"
REFRESH_TOKEN_SECRET="Your token secret"
MONGO_DB_URL = "Your test mongodb link"
TESTING_DB_URL="Your secondary mongodb link"
API_BASE= /api/v1
NODE_ENV="dev"
JWT_EXPIRATION=60
JWT_REFRESH_EXPIRATION=1200
SERVER_URL=http://127.0.0.1:3500
SERVER_NAME="LOCAL HOST"
```
#### Any background context you want to provide?
There are some issues that need working around:
- Invalidating **tokens before expiration**
#### What are the relevant pivotal tracker stories?
#180969915
#180969867
#### Screenshots (if appropriate)
Not provided
#### Questions:
No questions yet
